### PR TITLE
feat: add Mintable version of LSP7 / LSP8 

### PR DESCRIPTION
**Pull request by CJ42**
# What does this PR introduce?

- [x] create Mintable version of `LSP7DigitalAsset` and `LSP8IdentifiableDigitalAsset`

**NB: mintable only by the owner**

- [x] add unit tests for `LSP7Mintable` and `LSP8Mintable`

### CODE REVIEW

File: contracts/LSP7DigitalAsset/extensions/LSP7Mintable.sol [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+contract LSP7Mintable is LSP7MintableCore, LSP7DigitalAsset {
+    // solhint-disable no-empty-blocks
+    constructor(
+        string memory name_,
+        string memory symbol_,
+        address newOwner_,
+        bool isNFT_
+    ) LSP7DigitalAsset(name_, symbol_, newOwner_, isNFT_) {}
+
+    function mint(
```
Comment (frozeman) [inline contracts/LSP7DigitalAsset/extensions/LSP7Mintable.sol:20]: Why overwriting what you got in core?

File: contracts/LSP7DigitalAsset/extensions/LSP7Mintable.sol [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+
+import "./LSP7MintableCore.sol";
+import "../LSP7DigitalAsset.sol";
+
+/**
+ * @dev LSP7 extension, mintable.
+ */
+contract LSP7Mintable is LSP7MintableCore, LSP7DigitalAsset {
+    // solhint-disable no-empty-blocks
+    constructor(
```
Comment (frozeman) [inline contracts/LSP7DigitalAsset/extensions/LSP7Mintable.sol:13]: Why not inheriting from the constrictor version?

File: tests/LSP7DigitalAsset/extensions/LSP7Mintable.spec.ts [hunk 3]
```diff
@@ -1 +1,3 @@
-// TODO
\ No newline at end of file
+// TODO
+
+// test
```
Comment (Hugoo) [inline tests/LSP7DigitalAsset/extensions/LSP7Mintable.spec.ts]: Avoid pushing test commits ;)

File: tests/LSP7DigitalAsset/extensions/LSP7Mintable.behavior.ts [hunk 4]
```diff
@@ -1 +1,54 @@
-// TODO
\ No newline at end of file
+import { SignerWithAddress } from "@nomiclabs/hardhat-ethers/signers";
+import { ethers } from "hardhat";
+import { LSP7Mintable } from "../../../types";
+
+import type { BigNumber } from "ethers";
+
+export type LSP7MintableTestAccounts = {
```
Comment (Hugoo) [inline tests/LSP7DigitalAsset/extensions/LSP7Mintable.behavior.ts:5]: I think this type is not used anywhere else in your repo, if so, don't export it.

File: tests/LSP7DigitalAsset/extensions/LSP7Mintable.behavior.ts [hunk 5]
```diff
@@ -1 +1,54 @@
...
+      let postTotalSupply = await context.lsp7Mintable.totalSupply();
+      expect(postTotalSupply).toEqual(preTotalSupply.add(tokensToMint));
+    });
+
+    // it("owner balance should have increased", async () => {});
+  });
+
+  describe("when non-owner minting tokens", () => {
+    // it("a non-owner should not be allowed to mint", async () => {});
+  });
```
Comment (Hugoo) [inline tests/LSP7DigitalAsset/extensions/LSP7Mintable.behavior.ts]: ?

Add `// TODO:` comment maybe?

Or this: https://jestjs.io/fr/docs/api#testtodoname

File: tests/LSP7DigitalAsset/extensions/LSP7Mintable.behavior.ts [hunk 6]
```diff
@@ -1 +1,54 @@
...
+  buildContext: () => Promise<LSP7MintableTestContext>
+) => {
+  let context: LSP7MintableTestContext;
+
+  beforeEach(async () => {
+    context = await buildContext();
+  });
+
+  describe("when owner minting tokens", () => {
+    it("total supply should have increased", async () => {
```
Comment (Hugoo) [inline tests/LSP7DigitalAsset/extensions/LSP7Mintable.behavior.ts:38]: Maybe "When owner mints token, it should increase the total supply"

```
it("should increase the total supply"
```

File: tests/LSP8IdentifiableDigitalAsset/extensions/LSP8Mintable.behavior.ts [hunk 7]
```diff
@@ -1 +1,71 @@
-// TODO
\ No newline at end of file
+import { SignerWithAddress } from "@nomiclabs/hardhat-ethers/signers";
+import { ethers } from "hardhat";
+import { LSP8Mintable } from "../../../types";
+
+import type { BigNumber } from "ethers";
+
+export type LSP8MintableTestAccounts = {
```
Comment (Hugoo) [inline tests/LSP8IdentifiableDigitalAsset/extensions/LSP8Mintable.behavior.ts:5]: do you need to export this type?

File: tests/LSP7DigitalAsset/extensions/LSP7Mintable.behavior.ts [hunk 8]
```diff
@@ -0,0 +1,76 @@
...
+        amountToMint,
+        true, // beneficiary is an EOA, so we need to force minting
+        "0x"
+      );
+
+      let postTotalSupply = await context.lsp7Mintable.totalSupply();
+      expect(postTotalSupply).toEqual(preTotalSupply.add(amountToMint));
+    });
+
+    it("tokenReceiver balance should have increased", async () => {
```
Comment (Hugoo) [inline tests/LSP7DigitalAsset/extensions/LSP7Mintable.behavior.ts:53]: "it should increase the tokenReceiver balance" maybe?

File: tests/LSP8IdentifiableDigitalAsset/extensions/LSP8Mintable.behavior.ts [hunk 9]
```diff
@@ -0,0 +1,76 @@
...
+
+  beforeAll(async () => {
+    context = await buildContext();
+  });
+
+  describe("when owner minting tokens", () => {
+    it("total supply should have increased", async () => {
+      const randomTokenId = ethers.utils.randomBytes(32);
+
+      const preTotalSupply = await context.lsp8Mintable.totalSupply();
```
Comment (Hugoo) [inline tests/LSP8IdentifiableDigitalAsset/extensions/LSP8Mintable.behavior.ts:40]: maybe "preMintTotalSupply" and "postMintTotalSupply" ?

**Review by Hugoo — COMMENTED** [review]
cool

File: contracts/LSP7DigitalAsset/extensions/LSP7Mintable.sol (hunk 1 above)
Comment (CJ42) [reply contracts/LSP7DigitalAsset/extensions/LSP7Mintable.sol:20]: @frozeman overriding here to attach the `onlyOwner` modifier.

**Pull request merged**